classes/PrestaShopAutoload.php

```diff
@@ -81,14 +81,24 @@ public static function getInstance()
         return static::$instance;
     }
 
+    /**
+     * Get cache directory.
+     *
+     * @return string
+     */
+    protected static function getCacheDir(): string
+    {
+        return dirname(__DIR__) . '/var/cache';
```

Comment:
Ok, I can understand that you don't want to use config/environment variable for autoloading stuff (finding files under project defined directory tree).

But I really think hard coding cache directory isn't a good idea.
Cache is a dynamic directory that you may want to control when you are devops/admin (for example if you want to create some persistent volumes in a docker container). And the thing you don't want is to mix your sources with dynamic data.
For this question, Prestashop is already a mess. Removing `_PS_CACHE_DIR_` variable turns any refactoring even harder ...

Reply:
hardcoded path is needed, as this file is included before anything else, even before any configuration is/can be loaded

Reply:
If I understand well what you say, functions like this one 
```php
public static function getCacheFileIndex()
{
    return _PS_CACHE_DIR_ . 'class_index.php';
}
```
would search `class_index.php` in a wrong location during some process ?
Could you describe the use case where you have seen this behavior/bug ?

Reply:
please see dependent PR https://github.com/PrestaShop/PrestaShop/pull/28129/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34R182 - `classes/PrestaShopAutoload.php` is to be loaded before any config read, thus no user defined constants/config can be used

Reply:
Ok I got it.
So in fact you have anticipated a bootstrap process / architecture and your argue is that you have to implement this way cause of your goal.
Is it right to say that you want to remove environment variables so you have to hard code everything ?

Then I think we don't agree your goal/vision.

Why do you want to remove environment variable ?
Do you think Prestashop directory structure is right ? Once you have hard coded everything, what is your strategy to make it evolve ?
You didn't answer my implicit question about modern usages like Docker context with custom persistent volumes and read/write management issues ...

:thinking: 
Furthermore, I am thinking that we may not see Prestashop the same way neither :thinking:
What is Prestashop for you ? Do you see it as a standard website ?

Reply:
I am curious about the opinion of more experienced developers @PrestaShop/prestashop-core-developers  :slightly_smiling_face:

Reply:
> Is it right to say that you want to remove environment variables so you have to hard code everything ?

strictly hardcode only the path of autoloader index cache (AFAK only one file on generated on the 1st run (or if deleted after module install/uninstall))